plugins/inputs/opcua_listener/README.md

```diff
@@ -120,13 +150,24 @@ to use them.
   #   identifier_type = ""
   #   identifier = ""
   #
+  #   [inputs.opcua_listener.nodes.monitoring_params]
+  #     sampling_interval = "0s"
+  #     queue_size = "10"
+  #     discard_oldest = "true"
```

Comment:
This should be a boolean not a string; in toml you can do:

```toml
discard_oldest = true
```

Reply:
My intention here was to not break anything as well. The current implemention uses `true` for that parameter but if it is omitted in the config the value defaults to `false`. So, we don't know if that came from the user or just from the parameter not being present in the config. Maybe there is another way to check for the presence of a parameter in the config which I don't know of yet?

Reply:
> The current implemention uses true for that parameter but if it is omitted in the config the value defaults to false

When you say current implementation, you mean your PR? I don't see this in the repo today.

If the parameter does not exist when we unmarshall the config then we use the go default for a boolean, which is false. Is that not what you are trying to achive?

Reply:
With current implementation I mean how it currently works without my PR. There, function `NewMonitoredItemCreateRequestWithDefaults` of gopcua library is used which sets `true` for this parameter.

My intention was to keep this behaviour in case the parameter is not existent in the config. The best way I found was to use a string and check whether it's empty or not.

Reply:
Thanks for pointing out the default is set to true today. You can still set the default in the `func init()`

I would still like this to be a boolean in toml.

Reply:
I totally agree that a boolean would be the better choice in toml but I am struggling to find a solution to preserve the current behaviour in case the parameter is omitted in the config. As boolean parameters are automatically set to `false` when being omitted I cannot know whether this actually comes from a user's configuration or just because the parameter was omitted.

The following suggestions come to my mind (ordered in my personal preference):
- we define it as boolean and there is already a possibility which we don't have to implement first (maybe you know one?) to check the presence of a parameter in the config
- we leave it as it is in the PR
- we define it as boolean, assign the value anyways and therefore change the current default behaviour in case the parameter is omitted
- we leave it as string and change the possible values to e.g. "yes" and "no" so that user's don't accidentially use booleans

What do you think?